backend/inspirehep/records/marshmallow/literature/bibtex.py

```diff
@@ -79,14 +79,19 @@ def get_document_type(data, doc_type):
             "article": "article",
             "book": "book",
             "book chapter": "inbook",
-            "conference paper": "inproceedings",
             "proceedings": "proceedings",
             "report": "techreport",
             "note": "article",
         }
 
         if doc_type in DOCUMENT_TYPE_MAP:
             return DOCUMENT_TYPE_MAP[doc_type]
+        elif doc_type == "conference paper" and any(
+            [entry.get("journal_title") is not None
+             for entry in get_value(data, "publication_info")]):
```

Comment:
* you can access nested fields with `get_value` (that's the whole point):
```suggestion
        elif doc_type == "conference paper" and get_value(data, "publication_info.journal_title"):
```
* this `if`...`elif` chain is starting to get a bit unwieldy. The general pattern seems to be: either the type is static, or the type depends on the data. So I would suggest allowing the argument of the mapping to be either a string or a function, so you could have
```python
"conference paper": lambda data: "article" if get_value(data, "publication_info.journal_title") else "inproceedings"
```
You can also move `DOCUMENT_TYPE_MAP` to the class scope (and change to lower-case, this is a leftover from previous refactoring) so it's doesn't have to be evaluated for every record.